libraries/protocol/fee_schedule_calc.cpp

```diff
@@ -61,7 +61,7 @@ namespace graphene { namespace protocol {
    template<>
    uint64_t calc_fee_visitor::operator()(const htlc_create_operation& op)const
    {
-      transfer_operation::fee_parameters_type t;
+      transfer_operation::fee_parameters_type t = param.get<transfer_operation>();
```

Comment:
This doesn't look correct.
Please check `fee_helper` in `fee_schedule.hpp` for ideas about using param of another operation.
Actually I found `class fee_helper<htlc_create_operation>` there doing the default thing (the code was unnecessary).

Reply:
I am sorry, I am evidently missing something. 

As I am unable to add a parameter to htlc_create_operation::fee_parameters_type without messing up serialization, I am attempting to only retrieve 1 value from the transfer_operation::fee_parameters_type. I then want to include that one value in the htlc_create_operation.calculate_fee method. By calling param.get<transfer_operation>, I believe I am using the fee_helper to get me the correct transfer_operation::fee_parameters_type. 

Hopefully from my description of how I am understanding the issue, you can spot the part of the story I am missing.

Reply:
You're correct. The only issue here is `param.get<transfer_operation>()` may throw an exception if not found. On the other hand, `param.get<htlc_create_operation>()` won't throw due to the code I thought was unnecessary.

Reply:
So the question becomes do we specialize for `transfer_operation` or do we let it throw? In other words, if someone updates the fee schedule and removes the transfer_operation object, is that an exception, or should it go back to the defaults? My gut tells me that someone should be woken up, as there is a problem on the chain. But perhaps the intent of such an update would be to go back to the defaults.

Reply:
If a special fee param object is removed from the fee schedule, the default values will be used in other routines in the program. The code was a bit messy. I've created an issue https://github.com/bitshares/bitshares-core/issues/2150 for future improvements, but not a must in this release.

Reply:
@jmjatlanta please fix this, and resolve the conflicts, then I think we can merge the PR. Thanks.

Reply:
Perhaps I was not clear. When I was saying "please fix this", I mean only to fix the "param.get<transfer_operation>() may throw an exception" issue, but not a code refactory.